### ISSUE TITLE
Make images and albums selectable in the photowall

### ISSUE THREAD

**Issue by oparoz** [nextcloud/gallery#105]
_From @arkascha on January 3, 2016 20:56_

The file list view of a folder offers a checkbox per file to select multiple files and use actions on a set of selected files. 

The same would be very useful in the galleries photowall, since it offers a much better preview of the individual files. For albums this would clearly the preferred way to make a selection. Possible actions for sets of selected pictures would be: 
 - download of set as an archive
 - preview of set in slideshow
 - deletion of elements in set
 - moving of elements in set
 - ...

The actions are already implemented, this enhancement is just a duplication of an existing feature inside the front end.

_Copied from original issue: owncloud/gallery#487_

**Comment by oparoz** [comment]
Yes, it's on my roadmap, but after [drag and drop](https://github.com/owncloud/gallery/issues/405) has been implemented (and after that, there may be other priorities as well)

**Comment by desaintmartin** [comment]
Hello,
drag&drop has been merged, did the priorities change? ;)
If so, I might try to hack around to implement it very quickly (and dirty) for my own use case.

**Comment by oparoz** [comment]
Please have a go at it :)

**Comment by desaintmartin** [comment]
Would some common code with Files be preferable? For the moment I am making a lot of duplication, but I don't know the policy of sharing code between (even official) plugins.

I have some (ugly) working code on my instance, I may need a few days to clean this up a bit to be usable as a base to finish correct implementation.

**Comment by oparoz** [comment]
The best way is to duplicate libraries and code. We've had issues in the past where code changes in Files would then break Gallery.

**Comment by desaintmartin** [comment]
Sorry for the lack of updates, I'm still trying to continue despite lack of free time.

I am working on the feature for shared albums. I was wondering one thing: where does the files_sharing app kicks in when viewing shared directories in the files app? I've seen some related code in core/routes.php that loads all the app if it is defined but I am not sure WHERE it is first defined. I would then mimic this way of doing things, at first by just directly calling files_sharing (that monkey patch a lot of functions!) to see if it works.

**Comment by desaintmartin** [comment]
I've packed up an early working version of the feature, what do you think about it?

**Comment by desaintmartin** [comment]
I've packed-up an early working code on my fork :
https://github.com/desaintmartin/gallery/tree/feature-selection-wip

Beware : not tested yet.

What do you think about it, functionnally speaking? I propose first to look at what it should look like from the user perspective, then when it is ok I try (with your guidance on some points) to clean the code, separate from Files and properly test it.
